momo_modules/multi_sensor_module/src/app_main.c

```diff
@@ -2,52 +2,75 @@
 
 #include "platform.h"
 #include "watchdog.h"
-#include "sensor_defines.h"
 #include "port.h"
 #include "sample.h"
 #include "mib12_api.h"
 #include "log.h"
 #include "digital_amp.h"
 #include "watchdog.h"
+#include "ioc.h"
 #include "state.h"
 #include "pulse.h" 
 #include "alarm_repeat_times.h"
 
 MultiSensorState state;
+
 extern unsigned int adc_result;
+extern uint16_t counter;
+extern uint16_t periods;
 
 void task(void)
 {
-	while (state.acquire_pulse)
-	{
+	wdt_disable();
+
+	//If we slept for an increment, sample the number of pulses
+	if (nTO==0)
 		pulse_sample();
-		state.acquire_pulse = 0;
 
-		if ( state.push_pending )
-		{
-			state.push_pending = 0;
+	if (state.push_pending)
+	{
+		uint16_t correction_factor = 1;
+		state.push_pending = 0;
+
+		/*
+		 * We report data every minute, and we sample for 0.1 seconds
+		 * per time.  So we need to correct our sampled counter for the
+		 * ratio between the period of time that we are covering (60 seconds)
+		 * and the amount of time we were sampling for (periods*0.1 seconds)
+		 */
+
+		if (periods > 0)
+			correction_factor = 600/periods;
+
+		counter *= correction_factor;
 
-			bus_master_begin_rpc();
-			mib_buffer[0] = mib_address;
-			mib_buffer[1] = 0;
+		bus_master_begin_rpc();
+		mib_buffer[0] = mib_address;
+		mib_buffer[1] = 0;
 
-			mib_buffer[2] = 0;
-			mib_buffer[3] = 0; //metadata
+		mib_buffer[2] = 0;
+		mib_buffer[3] = 0; //metadata
 
-			mib_buffer[4] = pulse_count() & 0xFF;
-			mib_buffer[5] = (pulse_count() >> 8) & 0xFF;
-			mib_buffer[6] = 0;
-			mib_buffer[7] = 0;
+		mib_buffer[4] = counter & 0xFF;
+		mib_buffer[5] = counter & 0xFF;
+		mib_buffer[6] = 0;
```

Comment:
(future enhancement) Eventually the sensor event log API should be intelligent enough to accept a time interval for particular readings from a sensor module.